flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/v2/FilesCommitter.java

```diff
@@ -0,0 +1,224 @@
+    for (CommitRequest<FilesCommittable> request : requests) {
+      WriteResult committable = request.getCommittable().committable();
+      if (request.getNumberOfRetries() > 0) {
+        restored.add(request);
+        dataFilesNumRestored = dataFilesNumRestored + committable.dataFiles().length;
+        deleteFilesNumRestored = deleteFilesNumRestored + committable.deleteFiles().length;
+      } else {
+        store.add(request);
+        dataFilesNumStored = dataFilesNumStored + committable.dataFiles().length;
+        deleteFilesNumStored = deleteFilesNumStored + committable.deleteFiles().length;
+      }
+    }
+
+    if (restored.size() > 0) {
+      commitResult(dataFilesNumRestored, deleteFilesNumRestored, restored);
+    }
+
+    commitResult(dataFilesNumStored, deleteFilesNumStored, store);
+  }
+
+  @Override
+  public void close() throws Exception {
+    if (tableLoader != null) {
+      tableLoader.close();
+    }
+
+    if (workerPool != null) {
+      workerPool.shutdown();
+    }
+  }
+
+  private void commitResult(int dataFilesNum, int deleteFilesNum,
+                            Collection<CommitRequest<FilesCommittable>> committableCollection) {
+    int totalFiles = dataFilesNum + deleteFilesNum;
+
+    continuousEmptyCheckpoints = totalFiles == 0 ? continuousEmptyCheckpoints + 1 : 0;
+    if (totalFiles != 0) {
+      if (replacePartitions) {
+        replacePartitions(committableCollection, deleteFilesNum);
+      } else {
+        commitDeltaTxn(committableCollection, dataFilesNum, deleteFilesNum);
+      }
+
+      continuousEmptyCheckpoints = 0;
+    }
+  }
+
+  private void replacePartitions(Collection<CommitRequest<FilesCommittable>> writeResults, int deleteFilesNum) {
+    // Partition overwrite does not support delete files.
+    Preconditions.checkState(deleteFilesNum == 0, "Cannot overwrite partitions with delete files.");
+
+    // Commit the overwrite transaction.
+    ReplacePartitions dynamicOverwrite = table.newReplacePartitions().scanManifestsWith(workerPool);
+
+    int numFiles = 0;
+    String flinkJobId = "";
+    for (CommitRequest<FilesCommittable> writeResult : writeResults) {
+      WriteResult result = writeResult.getCommittable().committable();
+      Preconditions.checkState(result.referencedDataFiles().length == 0, "Should have no referenced data files.");
+      flinkJobId = writeResult.getCommittable().jobID();
+      numFiles += result.dataFiles().length;
+      Arrays.stream(result.dataFiles()).forEach(dynamicOverwrite::addFile);
+    }
+
+    commitOperation(dynamicOverwrite, numFiles, 0, "dynamic partition overwrite", flinkJobId);
+  }
+
+  private void commitDeltaTxn(Collection<CommitRequest<FilesCommittable>> writeResults, int dataFilesNum,
+                              int deleteFilesNum) {
+    if (deleteFilesNum == 0) {
+      // To be compatible with iceberg format V1.
+      AppendFiles appendFiles = table.newAppend().scanManifestsWith(workerPool);
+      String flinkJobId = "";
```

Comment:
nit: It is really not that important, but why not initialize the value with `null`? `""` just an unnecessary  constant